### DISCUSSION TITLE
Port typescript compiler to rust

### DISCUSSION THREAD

**Discussion by avin-kavish** [swc-project/swc#5372] [Ideas]
Would be nice to get that speed boost in type checking when it’s needed.

Also, will allow creation of compiler plugins like code transformation and reflection.

**Comment by kwonoj** [comment]
There is existing discussion https://github.com/swc-project/swc/issues/571

And also work in progress, status will be shared if there's something available to share. Please avoid creating duplicated discussions.

**Comment by kwonoj** [comment]
I locked discussion same as original issue as we experienced high volume of messages core team cannot follow in time. We will share progress once it's available.
